DataManagementSystem/Agent/FTS3Agent.py

```diff
@@ -71,11 +66,31 @@ def initialize(self):
     self.maxFilesPerJob = self.am_getOption("MaxFilesPerJob", 100)
     self.maxAttemptsPerFile = self.am_getOption("maxAttemptsPerFile", 256)
 
+    return S_OK()
+
+  def initialize(self):
+    """ agent's initialization """
+    self.fts3db = FTS3DB()
+    self._globalContextCache = {}
+
     # name that will be used in DB for assignment tag
     self.assignmentTag = gethostname().split('.')[0]
 
+    res = self.readConf()
+    if not res['OK']:
```

Comment:
given that `readConf` (that you can make private/protected btw) only returns `S_OK` we can maybe avoid it.